src/imap.rs

```diff
@@ -1063,6 +1063,16 @@ impl Session {
             .await
             .context("Failed to SELECT from imap_send")?
         {
+            // <https://www.rfc-editor.org/rfc/rfc3501#section-6.3.11>:
+            // If the mailbox is currently selected, the normal new message
+            // actions SHOULD occur. Specifically, the server SHOULD notify the
+            // client immediately via an untagged EXISTS response. If the server
+            // does not do so, the client MAY issue a NOOP command (or failing
+            // that, a CHECK command) after one or more APPEND commands.
+            //
+            // So deselect the folder just in case if the server doesn't send EXISTS. Then
```

Comment:
> Does it actually happen that server does not send `EXISTS`? Which server it is?

No, at least with chatmail servers.

IMU this RFC paragraph isn't about EXISTS related to APPEND itself, but about unsolicited EXISTS responses while APPEND is in progress. As for second device, i think any reasonable server implementation should notify it for appended messages too.

Reply:
There's a similar paragraph for the CHECK command:
```
      There is no guarantee that an EXISTS untagged response will happen
      as a result of CHECK.  NOOP, not CHECK, SHOULD be used for new
      message polling.
```
But i don't see such warnings for other commands. Also see https://www.rfc-editor.org/rfc/rfc3501#section-7:
```
   An example of unilateral untagged server data occurs when the IMAP
   connection is in the selected state.  In the selected state, the
   server checks the mailbox for new messages as part of command
   execution.  **Normally**, this is part of the execution of every command;
   hence, a NOOP command suffices to check for new messages.  If new
   messages are found, the server sends untagged EXISTS and RECENT
   responses reflecting the new size of the mailbox.
```
So i'm afraid this is about some strange implementations that don't send untagged EXISTS in all commands and we shouldn't rely on that. There's nothing too bad with closing the folder before APPEND, anyway the next step is `store_seen_flags_on_imap()` which may select another folder, so maybe it's not worth optimising here.

Reply:
I removed closing the folder before APPEND, but left uploading sync messages only from the inbox loop because if the mvbox for some reason can't be created and thus the mvbox loop doesn't exists, sync messages must be uploaded from the inbox loop anyway. And in this case they will be uploaded to Inbox.